Don't check for libsoup-2.4, as using its headers leads to broken compilation.
It tried it only because the 2.2.10x series for a short time was installed as a new 2.4 in development.
The real final libsoup-2.4 has API changes that don't work with the usage in this version (2.12.x) of evolution.

--- configure.in.orig	2008-04-01 21:18:44.000000000 +0300
+++ configure.in	2008-04-01 21:28:09.000000000 +0300
@@ -1518,14 +1518,11 @@
 AC_SUBST(LIBNOTIFY_LIBS)
 
 dnl --- evolution-calendar flags
-PKG_CHECK_MODULES(SOUPTEMP, libsoup-2.4, have_libsoup_24="yes", have_libsoup_24="no")
-if test $have_libsoup_24 = yes; then
-    LIBSOUP_REQUIRED=2.3.0
-    LIBSOUP=libsoup-2.4
-else
-    LIBSOUP_REQUIRED=2.2.2
-    LIBSOUP=libsoup-2.2
-fi
+dnl AC_ARG_VAR is used for keeping the diff against configure small
+AC_ARG_VAR(SOUPTEMP_CFLAGS)
+AC_ARG_VAR(SOUPTEMP_LIBS)
+LIBSOUP_REQUIRED=2.2.2
+LIBSOUP=libsoup-2.2
 
 EVO_SET_COMPILE_FLAGS(LIBSOUP, $LIBSOUP >= $LIBSOUP_REQUIRED)
 AC_SUBST(LIBSOUP_CFLAGS)
--- configure.orig	2008-04-01 21:18:47.000000000 +0300
+++ configure	2008-04-01 21:28:38.000000000 +0300
@@ -1989,9 +1989,9 @@
   LIBNOTIFY_LIBS
               linker flags for LIBNOTIFY, overriding pkg-config
   SOUPTEMP_CFLAGS
-              C compiler flags for SOUPTEMP, overriding pkg-config
+
   SOUPTEMP_LIBS
-              linker flags for SOUPTEMP, overriding pkg-config
+
   LIBSOUP_CFLAGS
               C compiler flags for LIBSOUP, overriding pkg-config
   LIBSOUP_LIBS
@@ -37358,81 +37358,9 @@
 
 
 
-pkg_failed=no
-{ echo "$as_me:$LINENO: checking for SOUPTEMP" >&5
-echo $ECHO_N "checking for SOUPTEMP... $ECHO_C" >&6; }
 
-if test -n "$PKG_CONFIG"; then
-    if test -n "$SOUPTEMP_CFLAGS"; then
-        pkg_cv_SOUPTEMP_CFLAGS="$SOUPTEMP_CFLAGS"
-    else
-        if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libsoup-2.4\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libsoup-2.4") 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_SOUPTEMP_CFLAGS=`$PKG_CONFIG --cflags "libsoup-2.4" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
-    fi
-else
-	pkg_failed=untried
-fi
-if test -n "$PKG_CONFIG"; then
-    if test -n "$SOUPTEMP_LIBS"; then
-        pkg_cv_SOUPTEMP_LIBS="$SOUPTEMP_LIBS"
-    else
-        if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libsoup-2.4\"") >&5
-  ($PKG_CONFIG --exists --print-errors "libsoup-2.4") 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_SOUPTEMP_LIBS=`$PKG_CONFIG --libs "libsoup-2.4" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
-    fi
-else
-	pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        SOUPTEMP_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libsoup-2.4"`
-        else
-	        SOUPTEMP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libsoup-2.4"`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$SOUPTEMP_PKG_ERRORS" >&5
-
-	have_libsoup_24="no"
-elif test $pkg_failed = untried; then
-	have_libsoup_24="no"
-else
-	SOUPTEMP_CFLAGS=$pkg_cv_SOUPTEMP_CFLAGS
-	SOUPTEMP_LIBS=$pkg_cv_SOUPTEMP_LIBS
-        { echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-	have_libsoup_24="yes"
-fi
-if test $have_libsoup_24 = yes; then
-    LIBSOUP_REQUIRED=2.3.0
-    LIBSOUP=libsoup-2.4
-else
-    LIBSOUP_REQUIRED=2.2.2
-    LIBSOUP=libsoup-2.2
-fi
+LIBSOUP_REQUIRED=2.2.2
+LIBSOUP=libsoup-2.2
 
 
 
